src/webapp/components/DeliveryDialog/index.js

```diff
@@ -1,23 +1,48 @@
-import React, { useState } from "react";
+import React, { useState, useContext } from "react";
+import axios from "axios";
 import { useTranslation } from "react-i18next";
 import Dialog from "@material-ui/core/Dialog";
+import DeliveryContext from "../../context/DeliveryContext";
 import {
   InfoStep,
   InstructionsStep,
   FormStep,
   FinishStep,
+  ErrorMessage,
 } from "./DialogScreens";
 
-const ClaimDeliveryDialog = ({ open, onClose, requestCode }) => {
+const ClaimDeliveryDialog = ({ open, onClose }) => {
+  const { requestCode } = useContext(DeliveryContext);
   const { t: str } = useTranslation();
   const [activeStep, setStep] = useState("info");
   const [phoneNumber, setPhoneNumber] = useState("");
+  const [error, setError] = useState();
 
-  const onSubmit = () => {
-    // mock api request
+  const onSubmit = async () => {
+    try {
+      await axios.post(`/api/delivery-needed/assign`, {
+        requestCode,
+        phoneNumber,
+      });
+      setStep("finish");
+    } catch ({ message }) {
+      if (message.includes("404"))
+        setError(
+          `We're sorry! Something went wrong while processing your request. Things that could have gone wrong include: we couldn't find the delivery request code, you haven't registered as a volunteer, or you entered the wrong phone number. Please contact #tech_support.`
```

Comment:
would be nice to surface the exact error in the future, but not blocking!

Reply:
same! I haven't quite figured out how to bubble up the server error message. Welcoming any tips 👀

Reply:
it seems like you are already returning the error message in the 404 body, maybe we can just use that to display the error instead?
<img width="1549" alt="Screen Shot 2020-08-29 at 7 11 11 PM" src="https://user-images.githubusercontent.com/1868400/91647695-8ee3ab80-ea2b-11ea-8efc-7b4bbf50b97c.png">

Reply:
i tried it out and you can extract the body of the error from axious as such!

patch for u here

```

diff --git a/src/webapp/components/DeliveryDialog/index.js b/src/webapp/components/DeliveryDialog/index.js
index b3699ee..5f1bb0d 100644
--- a/src/webapp/components/DeliveryDialog/index.js
+++ b/src/webapp/components/DeliveryDialog/index.js
@@ -25,11 +25,24 @@ const DeliveryDialog = ({ open, onClose, fetchData }) => {
         phoneNumber,
       });
       setStep("finish");
-    } catch ({ message }) {
+    } catch (err) {
+      const {
+        message,
+        response: {
+          data: { message: serverReadableError },
+        },
+      } = err;
       if (message.includes("404"))
-        setError(
-          `We're sorry! Something went wrong while processing your request. Things that could have gone wrong include: we couldn't find the delivery request code, you haven't registered as a volunteer, or you entered the wrong phone number. Please contact #tech_support.`
-        );
+        if (serverReadableError) {
+          setError(
+            `We're sorry! Something went wrong while processing your request: ${serverReadableError}. Please contact #tech_support.`
+          );
+        } else {
+          setError(
+            `We're sorry! Something went wrong while processing your request. Things that could have gone wrong include: we couldn't find the delivery request code, you haven't registered as a volunteer, or you entered the wrong phone number.
+            Please contact #tech_support.`
+          );
+        }
       else
         setError(
           `We're sorry! Something went wrong while processing your request. Please contact #tech_support and include the delivery request code and phone number you provided in your message.`
```

Reply:
oh sick thank you!! 🤩